pinot-query-runtime/src/main/java/org/apache/pinot/query/service/dispatch/QueryDispatcher.java

```diff
@@ -231,6 +232,9 @@ private static List<DataBlock> reduceMailboxReceive(MailboxReceiveOperator mailb
             "Received error query execution result block: " + transferableBlock.getDataBlock().getExceptions());
       }
       if (transferableBlock.isNoOpBlock()) {
+        // If all mailbox return no-op block, then we sleep for some time before retrying. This can increase the latency
+        // of the query a little bit but it will prevent the broker from running into a busy-waiting loop.
+        Thread.sleep(10);
```

Comment:
the best way to run this is a "pipelinebreaker" which is described as MVP in #10779 .

Reply:
Until we have that approach finalized, can we merge this to fix master?

We can revert this and fix this properly with the pipeline breaker PR?